include/ff/parallel_execution_ff.h

```diff
@@ -368,8 +369,15 @@ auto parallel_execution_ff::divide_conquer(Input & input,
     return condition_op(in);
   };
   output_type out_var{};
-  // divide, combine, seq, condition, input, res, wrks
-  ff::ff_DC<Input,output_type> dac(divide_fn, combine_fn, seq_fn, cond_fn, input, out_var, concurrency_degree_);
+
+  using dac_t = ff::ff_DC<Input,output_type>;
+  auto ncores = static_cast<int>(std::thread::hardware_concurrency());
+  int max_nworkers = std::max(concurrency_degree_, ncores);
```

Comment:
Why?
This means that if `my concurrency_degree_` is 2 but my number of cores is 8, then I would be running my d/c pattern with 8 workers.

Is that right?

Reply:
Sorry maybe the code was a bit unclear, I added some comments.

No, the number of workers is `concurrency_degree_`, whereas `max_nworkers` and `dac_t::DEFAULT_OUTSTANDING_TASKS` are ff-specific parameters for internal sizing of data structures.

I am not 100% sure that the ff API for the divide-and-conquer could not be simplified, but in its current version it needs to take both parameters.

Reply:
OK. I'm not a expert in FF. It seems to me that this implies that there will be never more threads than hardware concurrency even if the user really wants to perform oversubscription.

However, we accept the pull request until a better solution is found.

Reply:
No, the number of threads is `concurrency_degree_`, that is passed as parameter. If It is greater than the number of cores, there will be more threads than cores.

The `max_workers` parameter has no meaning from the user perspective, it is  there just for ff-internal purpose.